heartbeat/nagios4

```diff
@@ -0,0 +1,606 @@
+: ${OCF_RESKEY_cgidir="/usr/lib64/nagios/cgi-bin"}
+NAGIOS_BINDIR=`dirname ${OCF_RESKEY_binary}`
+: ${OCF_RESKEY_config="/etc/nagios/nagios.cfg"}
+: ${OCF_RESKEY_statusfile="/var/nagios/status.dat"}
+: ${OCF_RESKEY_statuslog="/var/nagios/livestatus.log"}
+: ${OCF_RESKEY_tempfile="/var/nagios/nagios.tmp*"}
+: ${OCF_RESKEY_retentionfile="/var/nagios/retention.dat"}
+: ${OCF_RESKEY_commandfile="/var/nagios/cmd/nagios.cmd"}
+: ${OCF_RESKEY_tempfile="/var/nagios/nagios.tmp"}
+: ${OCF_RESKEY_plugindir="/usr/lib64/nagios/plugins"}
+: ${OCF_RESKEY_user="nagios"}
+: ${OCF_RESKEY_group="nagios"}
+: ${OCF_RESKEY_PidFile="/var/nagios/nagios.pid"}
+
+OCF_RESKEY_PidFile="/var/nagios/nagios.pid"
+
+#
+# ---------------------------- VARIABLE DECLARATION ---------------------------
+#
+# WHAT: enable oracle tracing.
+#export ORACLE_TRACE='T'
+#
+SH=/bin/bash
+# ---------------------------- FUNCTION DECLARATION ---------------------------
+# WHAT: define usage function.
+# WHY:  to advise the usage of this script.
+# NOTE: 
+nagios_usage() {
+  cat <<USAGEND
+    usage: $0 (start|stop|validate-all|meta-data|help|usage|monitor)
+    $0 manages a nagios instance as an OCF HA resource.
+    The 'start' operation starts the instance.
+    The 'stop' operation stops the instance.
+    The 'status' operation reports whether the instance is running
+    The 'monitor' operation reports whether the instance seems to be working
+    The 'validate-all' operation reports whether the parameters are valid
+USAGEND
+}
+
+# WHAT: Define meta data to describe  Nagios service for pacemaker.
+# WHY:  
+nagios_meta_data() {
+        cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="nagios">
+<version>0.75</version>
+
+<longdesc lang="en">OCF Resource script for nagios 3.x or 4.x. It manages a nagios instance as an HA resource.</longdesc>
+<shortdesc lang="en">nagios resource agent</shortdesc>
+
+<parameters>
+
+<parameter name="installdir" unique="1" required="1">
+	<longdesc lang="en">Root directory of the nagios installation</longdesc>
+	<shortdesc lang="en">nagios install directory</shortdesc>
+	<content type="string" default="/usr/local/nagios" />
+</parameter>
+
+<parameter name="vardir" unique="1" required="1">
+	<longdesc lang="en">var directory of the nagios installation</longdesc>
+	<shortdesc lang="en">nagios var directory</shortdesc>
+	<content type="string" default="/usr/local/nagios/var" />
+</parameter>
+
+<parameter name="cgidir" unique="1" required="1">
+	<longdesc lang="en">cgi directory of the nagios installation</longdesc>
+	<shortdesc lang="en">nagios cgi directory</shortdesc>
+	<content type="string" default="/usr/local/nagios/sbin" />
+</parameter>
+
+<parameter name="binary" unique="1" required="1">
+	<longdesc lang="en">Location of the nagios binary</longdesc>
+	<shortdesc lang="en">nagios binary</shortdesc>
+	<content type="string" default="/usr/local/nagios/bin/nagios" />
```

Comment:
the default in the metadata does not match the default in the environment variables. Perhaps store defaults in some environment variables and use those in the metadata so you don't have to keep these synced.

default="$OCF_RESKEY_binary_default"